applications/ShapeOptimizationApplication/python_scripts/algorithms/algorithm_gradient_projection.py

```diff
@@ -270,9 +270,13 @@ def __getActiveConstraints(self):
     def __isConstraintActive(self, constraint):
         identifier = constraint["identifier"].GetString()
         constraint_value = self.communicator.getStandardizedValue(identifier)
-        if constraint["type"].GetString() == "=":
-            return True
-        elif constraint_value >= 0:
+        if constraint["type"].GetString() == "=" or constraint_value >= 0:
+            gradient_norm = self.optimization_utilities.ComputeMaxNormOfNodalVariable(
+                self.design_surface, self.constraint_gradient_variables[identifier]["mapped_gradient"]
+            )
+            if gradient_norm == 0.0:
+                KM.Logger.PrintWarning("ShapeOpt", f"Gradient for constraint {identifier} is 0.0 - will not be considered!")
```

Comment:
Is this gonna be a strict check, or should it be 
```suggestion
            if gradient_norm <= 1e-12:
                KM.Logger.PrintWarning("ShapeOpt", f"Gradient for constraint {identifier} is 0.0 - will not be considered!")
```

Reply:
Good point. From my experience 0.0 works in most cases since we really assign 0.0 in the sensitivity computation. But to be on the save side, we could also make it 1e-16 (i would prefer to make it very small)

Reply:
```suggestion
            if gradient_norm <= 1e-16:
                KM.Logger.PrintWarning("ShapeOpt", f"Gradient for constraint {identifier} is 0.0 - will not be considered!")
```